main.go

```diff
@@ -66,28 +68,43 @@ const (
 	webhookPort = 9443
 )
 
-func runPrometheusServer(metricsAddr string, tlsOptions common.SSPTLSOptions) error {
+func runPrometheusServer(metricsAddr string, ctx context.Context) error {
 	setupLog.Info("Starting Prometheus metrics endpoint server with TLS")
 	metrics.Registry.MustRegister(common_templates.CommonTemplatesRestored)
 	metrics.Registry.MustRegister(common.SSPOperatorReconcilingProperly)
 	handler := promhttp.HandlerFor(metrics.Registry, promhttp.HandlerOpts{})
 	mux := http.NewServeMux()
 	mux.Handle("/metrics", handler)
 
-	minTlsVersion, err := tlsOptions.MinTLSVersionId()
-	if err != nil {
-		return err
-	}
-
-	tlsConfig := tls.Config{
-		CipherSuites: tlsOptions.CipherIDs(&setupLog),
-		MinVersion:   minTlsVersion,
-	}
-
 	server := http.Server{
-		Addr:      metricsAddr,
-		Handler:   mux,
-		TLSConfig: &tlsConfig,
+		Addr:    metricsAddr,
+		Handler: mux,
+		TLSConfig: &tls.Config{
+			GetConfigForClient: func(_ *tls.ClientHelloInfo) (*tls.Config, error) {
+				cfg := &tls.Config{}
+				tlsOptions, err := common.GetSspTlsOptions(ctx)
+				if err != nil {
+					setupLog.Error(err, "Error while getting tls profile")
+					os.Exit(1)
+				}
+
+				if tlsOptions.IsEmpty() {
+					// if ciphersSuites is nil then default ciphers will be used
+					cfg.CipherSuites = nil
+					return cfg, nil
+				}
+				cfg.CipherSuites = tlsOptions.CipherIDs(&setupLog)
+				cfg.MinVersion, _ = minTLSVersionId(tlsOptions.MinTLSVersion)
+				certificate, err := loadCertificates(path.Join(sdkTLSDir, sdkTLSCrt), path.Join(sdkTLSDir, sdkTLSKey))
```

Comment:
Not sure how expensive it is to load certs on every request. You might cache some of this.

Reply:
Agreed.
There is already a certificate watcher in the `controller-runtime` library. Maybe we can use that, I will check...

Reply:
Here it is: https://github.com/kubevirt/ssp-operator/blob/9d3869accf1c183cdf66b3fdf572a3d8e9ec5a2d/vendor/sigs.k8s.io/controller-runtime/pkg/certwatcher/certwatcher.go#L45-L44

You can simply create a new `CertWatcher` and then add it to the `manager` to start it. Or call `Start()` manually in a separate goroutine.